### PR TITLE
Fixes #430, documentation about initial_types

### LINKED ISSUE #430: Understanding initial types
Could someone please shortly explain initial types?. Is there a way to programmatically infer the initial types from a given sklearn model?

### PR DESCRIPTION
**Pull request by xadupre**

### CODE REVIEW

File: docs/tutorial.rst [hunk 1]
```diff
@@ -72,3 +78,49 @@ for this machine learning model.
...
+*scikit-learn* does not store information about
+the training dataset. It is not always possible to retrieve
+the number of features or their types. That's why the
+function needs another argument called *initial_types*.
+In many cases, the training datasets is a numerical matrix
+*X_train*. Then it becomes
+``initial_type=[('X', FloatTensorType([None, X_train.shape[1]]))]``.
+*X* is the name of this unique input, the second term indicates the
+type and shape. The shape is ``[None, X_train.shape[1]]``,
+the first dimension is the number of rows, undefined, and the
```
Comment (prabhat00155) [inline docs/tutorial.rst:105]: Shouldn't we say `variable` instead of `undefined`?

File: docs/tutorial.rst [hunk 2]
```diff
@@ -72,3 +78,49 @@ for this machine learning model.
...
+
+::
+
+    initial_type=[
+        ('S', StringTensorType([None, 1])),
+        ('X', FloatTensorType([None, X_train.shape[1]])),
+    ]
+
+Function :func:`to_onnx <skl2onnx.to_onnx>` was implemented
+after discussions with the core developpers of *scikit-learn*.
```
Comment (prabhat00155) [inline docs/tutorial.rst:118]: Spelling: `developers`

Comment (xadupre) [reply docs/tutorial.rst]: fixed

File: docs/tutorial.rst (hunk 1 above)
Comment (xadupre) [reply docs/tutorial.rst]: I proposed a new formulation.

**Pull request merged**